### PR TITLE
Integration with Nebius AI Studio added

### PR DESCRIPTION
**Pull request by Aktsvigun**
Hey team,

Thanks for a great framework! Adding integration with Nebius AI Studio, which provides access to state-of-the-art models.

### CODE REVIEW

**Comment by Aktsvigun** [comment]
@ksolo may I kindly ask you to review it before it diverges too much from the main? thanks!

File: tests/providers/test_nebius_provider.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+    user_greeting = "We are testing you. Please say 'One two three' and nothing more."
+    message_history = [{"role": "user", "content": user_greeting}]
+    selected_model = "Qwen/Qwen2.5-32B-Instruct-fast"
+    chosen_top_p = 0.01
+    response_text_content = "One two three"
+
+    provider = NebiusProvider()
+    print(provider.api_key)
+    response = provider.chat_completions_create(model=selected_model, messages=message_history, top_p=chosen_top_p)
```
Comment (ksolo) [inline tests/providers/test_nebius_provider.py:17]: @Aktsvigun We have been mocking the interaction with the providers. If you look at an example like this one (https://github.com/andrewyng/aisuite/blob/main/tests/providers/test_sambanova_provider.py) we are mostly testing the setup more so than the actual result.

Can you update this to follow that pattern?

File: examples/client.ipynb
Comment (ksolo) [inline examples/client.ipynb:1]: I recommend resetting the notebook so the outputs are all clean when committing these.

**Review by ksolo — CHANGES_REQUESTED** [review]

File: examples/client.ipynb
Comment (Aktsvigun) [reply examples/client.ipynb:1]: Could you please specify what outputs you reference? In this notebook (examples/client.ipynb) I can see no outputs.

File: tests/providers/test_nebius_provider.py (hunk 1 above)
Comment (Aktsvigun) [reply tests/providers/test_nebius_provider.py]: Sure, fixed! Kindly check. 
I've standardized the test according to the example you sent & modified the NebiusProvider to match most of other providers (e.g. Mistral, Sambanova, etc.)

**Comment by ksolo** [comment]
@Aktsvigun apologies on the delay I was traveling for the holidays.

**Review by ksolo — APPROVED** [review]

**Comment by ksolo** [comment]
@Aktsvigun Can you open a follow up PR with documentation? Examples can be found here (https://github.com/andrewyng/aisuite/tree/main/guides)

**Comment by Aktsvigun** [comment]
Hi @ksolo, thank you! Sure, will do.

**Pull request merged**